pkg/controller/migplan/registry_health.go

```diff
@@ -0,0 +1,143 @@
+package migplan
+
+import (
+	"context"
+	"github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	"github.com/konveyor/mig-controller/pkg/compat"
+	health "github.com/konveyor/mig-controller/pkg/health"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	event2 "sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"time"
+)
+
+
+// A registry health checker which routinely checks whether registries are healthy or not
+type registryHealth struct {
+	hostClient client.Client
+	srcClient  client.Client
+	Interval   time.Duration
+	handler    handler.EventHandler
+	queue      workqueue.RateLimitingInterface
+	predicates []predicate.Predicate
```

Comment:
probably worth adding a `podSelector` field that we can pass from the instantiation. we can use this to watch other dependencies like velero and restic

Reply:
Added a new attribute `podLabels map[string]string`